packages/quantic/force-app/main/default/lwc/quanticTimeframeFacet/quanticTimeframeFacet.js

```diff
@@ -0,0 +1,526 @@
+   */
+  get showValues() {
+    return this._showValues;
+  }
+
+  /**
+   * Gets the specified timeframes.
+   * @returns {Timeframe[]} The specified timeframes.
+   */
+  get timeframes() {
+    /** @type {TimeframeElement[]} */
+    // @ts-ignore
+    const elements = Array.from(this.querySelectorAll('c-quantic-timeframe'));
+
+    return elements.map((el) => {
+      const amount =
+        typeof el.amount === 'string' ? parseInt(el.amount, 10) : el.amount;
+
+      return {
+        period: el.period,
+        unit: el.unit,
+        amount: amount,
+        label: el.label,
+      };
+    });
+  }
+
+  /**
+   * Gets the relative date ranges to request to the platform.
+   * @returns {DateRangeRequest[]} The relative date ranges.
+   */
+  get currentValues() {
+    return this.timeframes.map((timeframe) => {
+      return timeframe.period === 'past'
+        ? CoveoHeadless.buildDateRange({
+            start: {
+              period: timeframe.period,
+              unit: timeframe.unit,
+              amount: timeframe.amount,
+            },
+            end: {period: 'now'},
+          })
+        : CoveoHeadless.buildDateRange({
+            start: {period: 'now'},
+            end: {
+              period: timeframe.period,
+              unit: timeframe.unit,
+              amount: timeframe.amount,
+            },
+          });
+    });
+  }
+
+  /**
+   * Gets the facet values with the formatting applied.
+   */
+  get formattedValues() {
+    const values = this.facetState?.values || [];
+
+    return values
+      .filter((value) => value.numberOfResults > 0)
+      .map((value) => ({
+        ...value,
+        label: this.formatFacetValue(value),
+        key: JSON.stringify([value.start, value.end]),
+        selected: value.state === 'selected',
+      }));
+  }
+
+  get actionButtonIcon() {
+    return this.isCollapsed ? 'utility:add' : 'utility:dash';
+  }
+
+  get actionButtonCssClasses() {
+    return this.isCollapsed ? 'facet__expand' : 'facet__collapse';
+  }
+
+  get actionButtonLabel() {
+    const label = this.isCollapsed
+      ? this.labels.expandFacet
+      : this.labels.collapseFacet;
+    return I18nUtils.format(label, this.label);
+  }
+
+  get hasActiveValues() {
+    return (
+      this.formattedValues.some((v) => v.selected) || this.dateFilterState.range
+    );
+  }
+
+  get clearFilterLabel() {
+    return I18nUtils.format(this.labels.clearFilter);
+  }
+
+  get datepickerFormat() {
+    return I18nUtils.getShortDatePattern();
+  }
+
+  /**
+   * @returns {DatepickerElement|null}
+   */
+  get startDatepicker() {
+    // @ts-ignore
+    return this.withDatePicker
+      ? this.template.querySelector('.timeframe-facet__start-input')
+      : null;
+  }
+
+  /**
+   * @returns {DatepickerElement|null}
+   */
+  get endDatepicker() {
+    // @ts-ignore
+    return this.withDatePicker
+      ? this.template.querySelector('.timeframe-facet__end-input')
+      : null;
+  }
+
+  connectedCallback() {
```

Comment:
So far in all the other components the connectedCallback, renderedCallback and initialize functions are at the top of the file. Was it a conscious choice to put them here instead?

Reply:
The conscious choice was to put the properties before the functions. I don't mind moving the hooks to the top though.